### PR TITLE
Fix PortMapper issue on ZeroTier startup

### PR DESCRIPTION
**Pull request by laduke**
See issue #2082

We use a call to libnatpmp::ininatpp to make sure the computer has working network sockets before we go into the main nat-pmp/upnp logic.

With basic exponenetial delay up to 30 seconds.

### CODE REVIEW

**Comment by laduke** [comment]
OK. This addresses a PortMapper issue, but ZeroTier is still stuck. 

I think we need to add a similar check up in OneService or Binder or something...

**Comment by joseph-henry** [comment]
Updated to hopefully trigger the new workflow

**Comment by joseph-henry** [comment]
```
PortMapper: started for UDP port 20001
PortMapper: NAT-PMP: request for external address failed, aborting...
PortMapper: NAT-PMP: request failed, switching to UPnP mode
Starting Control Plane...
learned new path 103.195.103.66/9993 to 778cde7190 (packet 64b14547ff432878 local socket 90113408 network 0000000000000000)
learned new path 104.194.8.134/9993 to cafe9efeb9 (packet 64d0a85186548932 local socket 90113408 network 0000000000000000)
learned new path 84.17.53.155/9993 to cafe04eba9 (packet 63613cefe73921cc local socket 90113408 network 0000000000000000)
learned new path 50.7.252.138/9993 to 62f865ae71 (packet 64724b39ab70c2a7 local socket 90113632 network 0000000000000000)
PortMapper: upnpDiscover failed, returning to NAT-PMP mode: 0
```

**Comment by wongsyrone** [comment]
initnatpmp() will try to connect to `<gateway>:<NATPMP_PORT>` and fail with error code NATPMP_ERR_CONNECTERR if there is no such service on the gateway. In this case, your code treats the network as not available.

**Comment by wongsyrone** [comment]
https://github.com/zerotier/ZeroTierOne/blob/27a9cc6254304c318be0f1b75fddff47210c1d2f/ext/libnatpmp/natpmp.c#L97

File: osdep/PortMapper.cpp [hunk 1]
```diff
@@ -293,6 +313,7 @@ class PortMapperImpl
 #ifdef ZT_PORTMAPPER_TRACE
                     PM_TRACE("PortMapper: upnpDiscover failed, returning to NAT-PMP mode: %d" ZT_EOL_S,upnpError);
 #endif
+					break;
```
Comment (wongsyrone) [inline osdep/PortMapper.cpp:316]: This 'break' exits the `while (run)` loop

**Comment by joseph-henry** [comment]
Ok, so this passes tests and it does seem to make sense. I think we're ready for a merge

**Pull request merged**